src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -426,6 +426,19 @@ public DependencyGraphSpec WithoutTools()
             return newSpec;
         }
 
+        /// <summary>
+        /// Create a minimal DependencyGraphSpec that will pass restore. Typically used for error handling.
+        /// </summary>
+        /// <param name="projectPath">Full path to project file.</param>
+        /// <returns>DependencyGraphSpec</returns>
+        public static DependencyGraphSpec CreateMinimal(string projectPath, string outputPath)
```

Comment:
Not a fan of the factory methods in such a low level class :(
There's no use beyond what you are doing is there? 
I don't  think we enough use-cases to define a good API. 
Why can't it be private/internal to the assembly that creates them.